### ISSUE TITLE
DBC metadate not match the type

### ISSUE THREAD

**Issue by Wangmmx** [DeepBrainChain/DeepBrainChain-MainChain#84]
In DBC metadata, storage event Asset transferrd type is :
```
  {
                                "args": [
                                    "AssetId", 
                                    "AccountId", 
                                    "AccountId", 
                                    "Balance"
                                ], 
                                "name": "Transferred", 
                                "documentation": [
                                    " Some assets were transferred. \\[asset_id, from, to, amount\\]"
                                ]
                            }, 

```
But the type of asset amount is u64, so types can not decoded from metadata, is this a bug?

**Comment by Wangmmx** [comment]
> 

JOSNRPC :state_getMetadata， and endpoint is wss://info.dbcwallet.io

**Comment by Wangmmx** [comment]
@kaigedong maybe could you please help me with this?

**Comment by kaigedong** [comment]
I don't know how polkadot.js handle associated types(assets pallet and balance pallet both use [Balance](https://github.com/DeepBrainChain/DeepBrainChain-MainChain/blob/3199024b0022a412d77441ae6129b07357c27df3/runtime/src/lib.rs#L1007) ), and event can be [decoded successfully](https://www.dbcwallet.io/?rpc=wss%3A%2F%2Finfotest.dbcwallet.io%3A7777#/explorer/query/0x9b4627e74226ada4c9db2c24495037f8a5037c18b149a131faf69231c75c1f78). 

Maybe you can change dbc_types.json, add a line `"TAssetBalance": "u64"`. polkadot.js will use this to decode assets transfer event.

**Comment by Wangmmx** [comment]
@kaigedong could you please help me with this?

> I don't know how polkadot.js handle associated types(assets pallet and balance pallet both use [Balance](https://github.com/DeepBrainChain/DeepBrainChain-MainChain/blob/3199024b0022a412d77441ae6129b07357c27df3/runtime/src/lib.rs#L1007) ), and event can be [decoded successfully](https://www.dbcwallet.io/?rpc=wss%3A%2F%2Finfotest.dbcwallet.io%3A7777#/explorer/query/0x9b4627e74226ada4c9db2c24495037f8a5037c18b149a131faf69231c75c1f78).
> 
> Maybe you can change dbc_types.json, add a line `"TAssetBalance": "u64"`. polkadot.js will use this to decode assets transfer event.

Thanks for the reply first
I did not use the type in polkadot.js, I build and decode the transactions offline, I use a jsonrpc call `state_getMetadata`, got all the types, then decode transaction according to each type

And I found In DBC metadata,  event about Asset's amount are all type `Balance`, in both `Call `and `Storage` module,  I add the type but didn't work, Thanks again and maybe I will try some other endpoints to find the problem

**Comment by kaigedong** [comment]
Not all types info is stored in metadata, so you should provide more custom types (this repo is developed based on substrate v3.0.0, not all types info is stored in metadata.). For DBC, it's custom types is in https://github.com/DeepBrainChain/DeepBrainChain-MainChain/blob/master/dbc_types.json. For subscan, polkascan or polkadot.js, custom type is needed to decode events. So you can check the way they use to decode events. 

Besides, types like Balance, AccountId or something else is stored in node module `@polkadot/types`, for Balance type in assets module, they use TAssetBalance as asset Balance's alias. So that dbcwallet can decode assets.transfer event like https://www.dbcwallet.io/?rpc=wss%3A%2F%2Finfotest.dbcwallet.io%3A7777#/explorer/query/0x9b4627e74226ada4c9db2c24495037f8a5037c18b149a131faf69231c75c1f78

**Comment by Wangmmx** [comment]
Yes I have already added the custom types in https://github.com/DeepBrainChain/DeepBrainChain-MainChain/blob/master/dbc_types.json about 2 weeks ago, and update it everyday. 

Other events can be decoded correctly, only Asset modules failed cause there's no type named 'TAssetBalance' in metadata, so I can not replaced it in custom types, I will check the subscan's way to decode it and try it again, thanks for your help
